homeassistant/components/sensor/date_countdown.py

```diff
@@ -0,0 +1,70 @@
+"""
+Date Countdown
+For more details about this sensor please refer to the documentation at
+https://home-assistant.io/components/sensor.date_countdown/
+
+"""
+
+import datetime
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "Countdown"
+CONF_DATE = 'date'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_DATE): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up date countdown sensor."""
+    end_date = config.get(CONF_DATE)
+    sensor_name = config.get(CONF_NAME)
+
+    add_devices([Countdown(sensor_name, end_date)])
+
+
+class Countdown(Entity):
+    """Implementation of the date countdown sensor."""
+
+    def __init__(self, sensor_name, end_date):
+        """Initialize the sensor."""
+        self.end_date = end_date
+        self._name = sensor_name
+        self._state = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return 'mdi:calendar'
+
+    def update(self):
+        end_date = datetime.datetime.strptime(self.end_date, '%m-%d-%Y %H:%M')
```

Comment:
`%m-%d-%Y` is an US-thing. If you deviate from ISO 8601 then it should be `%d-%m-%Y` which is used by ten times more people than `%m-%d-%Y` in regard of the .